packages/nestjs-module/src/ogma-core.module.ts

```diff
@@ -42,16 +41,6 @@ export class OgmaCoreModule extends createConfigurableDynamicRootModule<
       useFactory: createOgmaServiceOptions,
       inject: [OGMA_OPTIONS],
     },
-    {
-      provide: APP_INTERCEPTOR,
-      useFactory: createOgmaInterceptorFactory,
-      inject: [
-        OGMA_INTERCEPTOR_OPTIONS,
-        OgmaService,
-        DelegatorService,
-        Reflector,
-      ],
-    },
```

Comment:
I assume you've removed this to put the necessity on the dev to attach the OgmaInterceptor?

Reply:
Yes, I assume that everyone who would like to use `OgmaInterceptor` need to inject it by themself by eg.

```ts
@Module({
  ...
  providers: [
    {
      provide: APP_INTERCEPTOR,
      useClass: OgmaInterceptor
    }
  ]
})
export class MyModule {}
```

or by overwriting `OgmaInceptor` and inject self created one

Reply:
This is a great addition. As nice as it is to have the interceptor applied automatically, having it be applied by the user is definitely going to clean some stuff up. We will need to make a note of this in the readme of course. and I believe you didn't change the module options, right? So if no adapter is applied and `false` is not set on the `interceptor` option there's still a problem, right?

Reply:
Yeah I didn't change that. I was trying to move option `interceptor` into constructor of `OgmaInterceptor` but that change was huuuuuge and turns out that need to be changed delegate service and few others concepts. I didn't want to change something that you wouldn't like as that change impact on almost whole package.

There is no option to not set interceptor in options as it's required by types. But yeah I didn't touch options on `forRoot` and `forAsyncRoot`

Reply:
Perfect. I think overall, I'd like to keep that part of the functionality the same, just because if an adapter wasn't registered to begin with, and the interceptor was bound, there could be all sorts of other errors. Still can be, but now it's less likely.